src/DistributedMetadata/DDLService.cpp

```diff
@@ -478,11 +504,25 @@ void DDLService::mutateDatabase(DWAL::RecordPtr record, const String & method) c
 
     std::vector<Poco::URI> target_hosts{toURIs(hosts, fmt::format(*api_path_fmt, database))};
 
+    std::vector<String> failed_hosts;
     /// FIXME : Parallelize doDDL on the uris
     for (auto & uri : target_hosts)
     {
         uri.setQueryParameters(Poco::URI::QueryParameters{{"distributed_ddl", "false"}});
-        doDDL(payload, uri, method, query_id, user);
+        auto err = doDDL(payload, uri, method, query_id, user);
+        if (err != ErrorCodes::OK)
+        {
+            failed_hosts.push_back(uri.getHost() + ":" + toString(uri.getPort()));
+        }
+    }
+
+    if (failed_hosts.empty())
+    {
+        succeedDDL(query_id, user, payload);
```

Comment:
Whether there is a logical conflict with line 528 ？

Reply:
good  cache.